src/main/java/org/jenkinsci/plugins/sshsteps/steps/CommandStep.java

```diff
@@ -68,7 +69,14 @@ protected Object run() throws Exception {
       if (Util.fixEmpty(step.getCommand()) == null) {
         throw new IllegalArgumentException("command is null or empty");
       }
-      return getLauncher().getChannel().call(new CommandCallable(step, getListener()));
+
+      final VirtualChannel channel = getLauncher().getChannel();
+      if (channel == null) {
+        throw new IllegalArgumentException(
+            "Unable to get the channel, Perhaps you forgot to surround the code with a step that provides this, such as: node, dockerNode");
```

Comment:
Can we add a helper method to encapsulate this rather than repeating this in each of the steps? Maybe in the base class?

Reply:
Addressed with https://github.com/jenkinsci/ssh-steps-plugin/pull/23/commits/8c24ed4484776edcb2e3e0df9d5d9a63fb34db38